external/avro/src/test/scala/org/apache/spark/sql/avro/AvroSerdeSuite.scala

```diff
@@ -86,18 +86,18 @@ class AvroSerdeSuite extends SparkFunSuite {
     // deserialize should have no issues when 'bar' is nullable but fail when it is nonnull
     Deserializer.create(CATALYST_STRUCT, avro, BY_NAME)
     assertFailedConversionMessage(avro, Deserializer, BY_NAME,
-      "Cannot find non-nullable field 'foo.bar' (at position 0) in Avro schema.",
+      "Cannot find field 'foo.bar' (at position 0) in Avro schema at field 'foo'",
```

Comment:
The wording `at field 'foo'` seems unnecessary, as we already show the full field name 'foo.bar'.
Also, "Cannot find **field** .. in Avro schema at **field** .." sounds odd.

Reply:
Yes I think you're right. I think the issue is that I was trying too hard to unify the code paths / exception messages for the by-name and by-position cases. I've updated this and I think the messages should be more readable now.

In the case of positional matching, I still report the field names of both the SQL and Avro fields, since they may be different. I can add logic to omit the second instance of the name if they're matching, but I don't think it's necessary. Lmk your thoughts.